src/vectorpostprocessors/ResponseHistoryMean.C

```diff
@@ -0,0 +1,81 @@
+// MOOSE includes
+#include "ResponseHistoryMean.h"
+#include "PostprocessorInterface.h"
+#include "VectorPostprocessorInterface.h"
+#include "MastodonUtils.h"
+#include "ResponseHistoryBuilder.h"
+/* #include <iostream>
+#include <fstream>
+using namespace std;*/
+
+registerMooseObject("MastodonApp", ResponseHistoryMean);
+
+template <>
+InputParameters
+validParams<ResponseHistoryMean>()
+{
+  InputParameters params = validParams<GeneralVectorPostprocessor>();
+  params.addRequiredParam<VectorPostprocessorName>(
+      "vectorpostprocessor",
+      "Name of the ResponseHistoryBuilder vectorpostprocessor, for which "
+      "response spectra are calculated.");
+
+  // Make sure that csv files are created only at the final timestep
+  params.set<bool>("contains_complete_history") = true;
+  params.suppressParameter<bool>("contains_complete_history");
+
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_FINAL};
+  params.suppressParameter<ExecFlagEnum>("execute_on");
+
+  params.addClassDescription("Calculate the mean acceleration time series given a set of nodes.");
+  return params;
+}
+
+ResponseHistoryMean::ResponseHistoryMean(const InputParameters & parameters)
+  : GeneralVectorPostprocessor(parameters),
+    _history_time(getVectorPostprocessorValue("vectorpostprocessor", "time"))
+
+{
+}
+
+void
+ResponseHistoryMean::initialSetup()
+{
+  const ResponseHistoryBuilder & history_vpp = getUserObjectByName<ResponseHistoryBuilder>(
+      getParam<VectorPostprocessorName>("vectorpostprocessor"));
+  std::vector<std::string> history_names =
+      history_vpp.getHistoryNames(); // names of the vectors in responsehistorybuilder
+  _history_acc.resize(history_names.size());
+
+  for (std::size_t i = 0; i < history_names.size(); i++)
+  {
+    // cout << history_names[i] << endl;
+    _history_acc[i] = history_vpp.getHistories()[i];
+  }
+
+  // Declaring a vector of pointers for storing the addresses of the time vector
+  // and the mean response history vector.
+
+  _mean_acc.push_back(&declareVector("_time"));
+  _mean_acc.push_back(&declareVector("_mean"));
+}
+
+void
+ResponseHistoryMean::initialize()
```

Comment:
You shouldn't need this, because this class is only executed on FINAL it only runs once so there is nothing to clear.

Reply:
I'm getting the following three errors when I exclude ResponseHistoryMean::initialize(). However, the code is working when I include initialize().

_**/opt/moose/llvm-8.0.0/bin/../include/c++/v1/memory:4325:5: error: static_assert failed due to requirement 'is_constructible<ResponseHistoryMean, const InputParameters &>::value'
      "Can't construct object in make_shared"**_

_**/opt/moose/llvm-8.0.0/bin/../include/c++/v1/memory:2168:7: error: field type 'ResponseHistoryMean' is an abstract class**_

_**/opt/moose/llvm-8.0.0/bin/../include/c++/v1/memory:4334:28: error: assigning to 'std::__1::__shared_weak_count *' from incompatible type
      'std::__1::unique_ptr<std::__1::__shared_ptr_emplace<ResponseHistoryMean, std::__1::allocator<ResponseHistoryMean> >,
      std::__1::__allocator_destructor<std::__1::allocator<std::__1::__shared_ptr_emplace<ResponseHistoryMean, std::__1::allocator<ResponseHistoryMean> > > > >::pointer' (aka
      'std::__1::__shared_ptr_emplace<ResponseHistoryMean, std::__1::allocator<ResponseHistoryMean> > *')**_

Reply:
I guess you need the method, but it can just be empty.

Reply:
Yes, it works when it is empty.